guessing_game.rb

```diff
@@ -0,0 +1,66 @@
+
+
+class GuessingGame
+
+  attr_accessor :congrats_message, :remaining_guesses
+
+
+  def initialize(secret_number,num_guesses)
+    @secret_number = secret_number
+    @num_guesses = num_guesses
+    @congrats_message = "Yay, you won!"
+    @remaining_guesses = @num_guesses
+    @has_won = false
+    @has_lost = false
+    @previous_guesses = []
```

Comment:
it will be faster to store the previous guesses in a hash. you can access a hash element in `O(1)` constant time, whereas searching an array for a particular element will require `O(n)` linear search